src/panels/lovelace/cards/hui-gauge-card.js

```diff
@@ -0,0 +1,198 @@
+          font-size: calc(var(--base-unit) * 0.55);
+        }
+        .gauge-data #title{
+          padding-top: calc(var(--base-unit) * 0.15);
+          font-size: calc(var(--base-unit) * 0.30);
+        }
+        .not-found {
+          flex: 1;
+          background-color: yellow;
+          padding: 8px;
+        }
+      </style>
+      <ha-card  on-click='_handleClick'>
+        <div class='container'>
+          <div class='gauge-a'></div>
+          <div class='gauge-b'></div>
+          <div class='gauge-c' id='gauge'></div>
+          <div class='gauge-data'><div id='percent'></div><div id='title'></div></div>
+        </div>
+      </ha-card>
+    `;
+  }
+
+  static get properties() {
+    return {
+      hass: {
+        type: Object,
+        observer: '_hassChanged'
+      },
+      _config: Object,
+    };
+  }
+
+  ready() {
+    super.ready();
+    if (this._config) this._buildConfig();
+  }
+
+  getCardSize() {
+    return 1;
+  }
+
+  setConfig(config) {
+    if (!config || !config.entity) throw new Error('Invalid card configuration');
+
+    this._config = Object.assign({ min: 0, max: 100 }, config);
+    if (this.$) this._buildConfig();
+  }
+
+  _buildConfig() {
+    const stateObj = this.hass.states[this._config.entity];
+    if (!stateObj) {
+      this.$.title.textContent = 'Entity not available: ' + this._config.entity;
+      this.$.title.className = 'not-found';
+      return;
+    }
+    if (this._config.attribute) {
+      stateObj.state = stateObj.attributes[this._config.attribute];
+    }
+    if (isNaN(stateObj.state)) {
+      this.$.title.textContent = 'Entity is non-numeric: ' + this._config.entity;
+      this.$.title.className = 'not-found';
+      return;
+    }
+
+    const unitOfMeasurement = this._config.unit_of_measurement || stateObj.attributes.unit_of_measurement || '';
+
+    if (!this.stateObj || stateObj.state !== this._stateObj.state) {
+      this.$.percent.textContent = `${stateObj.state} ${unitOfMeasurement}`;
+      this.$.title.textContent = this._config.title;
+      const turn = this._translateTurn(stateObj.state, this._config) / 10;
+      this.$.gauge.style.transform = `rotate(${turn}turn)`;
+      this.$.gauge.style.backgroundColor = this._computeSeverity(stateObj.state, this._config.severity);
+      this._stateObj = stateObj;
+    }
+  }
+
+  _hassChanged() {
+    this._buildConfig();
+  }
+
+  _computeSeverity(stateValue, sections) {
+    const numberValue = Number(stateValue);
+    const severityMap = {
+      red: 'var(--label-badge-red)',
+      green: 'var(--label-badge-green)',
+      amber: 'var(--label-badge-yellow)',
+      normal: 'var(--label-badge-blue)',
+    };
+    if (!sections) return severityMap.normal;
+    const sectionsArray = Object.keys(sections);
+
+    const sortable = sectionsArray.map((severity) => {
+      if (severityMap[severity] != null) {
+        return [severity, sections[severity]];
+      }
+      return severityMap.normal;
```

Comment:
The return value from this arrow function gets becomes an entry in the resulting array. If we hit this, one of the entries in `sortable` will just be a string so we'll get an out of bounds index below